### ISSUE TITLE
For triage: try cold page load with ublock in fennec, check for perf problems

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#17448]
From https://github.com/mozilla-mobile/fenix/issues/16473#issuecomment-737546522:

> Triage: we were able to reproduce with HTTPSEverywhere and ublock but not other add-ons. We still believe it's likely to be the code run by the add-ons (i.e. there's nothing we can do). One thing we can check is if we can reproduce the same delay [in fennec](https://archive.mozilla.org/pub/mobile/releases/68.9.0/). If so, then it's more evidence that it's due to code run by the add-ons and the issue should be filed against the specific add-ons. If not, we should file an issue against GeckoView because this is a regression.

Let's do this action – to check fennec for perf issues compared to fenix – as a separate issue and report back to the original issue linked above.

**Comment by ale82to** [comment]
ublock has a specific setting for speed up cold boot in advanced prefers.
works perfectly

**Comment by Mohan40** [comment]
I installed fennec from the above link and checked for the delay. There is **no delay** when the browser closed and opened again.

The webpage takes a second and normally loads without any long delay. I tried the above with uBlock Origin and HTTPS Everywhere. I remember using fennec before fenix was released and never experienced any delay in fennec. The issue started with the first version of fenix for me.

![Screenshot_20210115-151813_Firefox](https://user-images.githubusercontent.com/43498254/104797925-42d1b180-57e8-11eb-8142-d50b44f7233d.jpg)

**Comment by agi** [comment]
We've looked at this (again) with @csadilek yesterday, and we discovered some interesting things:

* The reason why Fenix startup is slower when running some extensions like uBlock Origin or HTTPS Everywhere is that these extensions look at all network requests (to decide whether to block them or not) and so the extension needs to startup before we can make any connection.

* Furthermore, these extensions have high startup time (presumably due to loading a high number of block rules).

* The reason why Fennec does not have a performance penalty when running uBlock or HTTPS everywhere is that it doesn't wait for extensions to be initialized before running network requests, which means that the first few network requests will be missed by the extension. You can easily reproduce this by loading a google results page and then restarting the browser, in Fennec the startup pageload will not have any elements blocked by uBlock, while subsequent ones will have `(2)` or whatever.

* In Fenix, using the profiler, the time spent is entirely inside the extension startup code, and looks like already somewhat optimized (HTTPS everywhere uses wasm for example), there doesn't seem to be anything obvious we can do to help here in the browser, short of making javascript faster in general.

**Comment by csadilek** [comment]
Here are the profiles we took while testing the first page load with and without HTTPS Everywhere. Note that absolute times are not relevant here. 

**Profile with HTTPS Everywhere installed:** 
https://share.firefox.dev/2Y2evIg

Here you see a significant amount of time spent running extension code in the parent process before the page load and content process starts: Here's the segment "zoomed in": https://share.firefox.dev/2Y3ZZzr 

**Profile without HTTPS Everywhere:** 
https://share.firefox.dev/2Kyh1CR

Here you see the content process and page load starting immediately.

The same pattern is reproducible with uBlock as well. Waiting for the extension to be fully initialized before the first page load is the correct behaviour. See Agi's comment above why this actually didn't work correctly in Fennec.

I will close this as it doesn't appear there's anything more we can do.
